crates/bevy_render/src/pipeline/render_pipelines.rs

```diff
@@ -82,6 +83,27 @@ pub fn draw_render_pipelines_system(
     meshes: Res<Assets<Mesh>>,
     mut query: Query<(&mut Draw, &mut RenderPipelines, &Handle<Mesh>)>,
 ) {
+    // Pipelines will be rebuilt for shaders that have been modified.
+    let mut changed_shaders = HashSet::default();
```

Comment:
I think we should move shader updating to its own system (like the current mesh_resource_provider_system)